pennylane_lightning/core/src/lightning_gpu.toml

```diff
@@ -0,0 +1,117 @@
+schema = 1
+
+[device]
+name = "lightning.qubit"
+precision = ['float32', 'float64']
+
+[operations]
+observables = [
+	"NamedObs",
+	"HermitianObs",
+	"TensorObs",
+	"HamiltonianObs",
+]
+
+# These are gates which are defined in Quantum Instruction Set.
+[[operations.gates]]
+native = [
+        "QubitUnitary",
+        "PauliX",
+        "PauliY",
+        "PauliZ",
+        "MultiRZ",
+        "Hadamard",
+        "S",
+        "T",
+        "CNOT",
+        "SWAP",
+        "CSWAP",
+        "Toffoli",
+        "CY",
+        "CZ",
+        "PhaseShift",
+        "ControlledPhaseShift",
+        "RX",
+        "RY",
+        "RZ",
+        "Rot",
+        "CRX",
+        "CRY",
+        "CRZ",
+        "CRot",
+        "IsingXX",
+        "IsingYY",
+        "IsingZZ",
+        "IsingXY",
+        # Adjoint'ed gates
+        # which just need the inverse flag set
+        # are considered natively supported gates.
+        "Adjoint(S)",
+        "Adjoint(T)",
+]
+# decomp gates are those which are neither supported natively
+# and are not executed via the applyMatrix operation. Technically
+# they could also be decomposed via applyMatrix.
+decomp = [
+        "SX",
+        "ISWAP",
+        "PSWAP",
+        "SISWAP",
+        "SQISW",
+        "CPhase",
+        # Adjoint'ed gates
+        # which are not in native set must either be decomposed
+        # or applied via matrix.
+        "Adjoint(SX)",
+        "Adjoint(ISWAP)",
+        "Adjoint(SISWAP)",
+]
+# QubitUnitary should never be here.
+# Gates in "matrix" are decomposed to QubitUnitary.
+# Having a QubitUnitary here would mean an infinite loop.
+
+# These are gates that appear on the list of supported gates
+# but are not in the list of Quantum Gate Set Instructions in
+# the Catalyst repository.
+# TODO: Move some of these to native once we add support for them
+# in Catalyst runtime.
+matrix = [
+        "Identity",
+        "BasisState",
+        "QubitStateVector",
+        "StatePrep",
+        "ControlledQubitUnitary",
+        "MultiControlledX",
+        "DiagonalQubitUnitary",
+        "SingleExcitation",
+        "SingleExcitationPlus",
+        "SingleExcitationMinus",
+        "DoubleExcitation",
+        "DoubleExcitationPlus",
+        "DoubleExcitationMinus",
+        "QubitCarry",
+        "QubitSum",
+        "OrbitalRotation",
+        "QFT",
+        "ECR",
+]
```

Comment:
All the gates here are in the list of allowed operations from the device. This means that they are not decomposed. Some of these are not implemented by `apply_matrix` like `SingleExcitation`... but we don't yet have support for them in the runtime. I.e., there is no runtime function `__qir__qis__SingleExcitation`. So at the moment they should be placed either in decomp or matrix. Once we implement them, we move them to `native` gates.

Reply:
Right, but what I'm wondering is how you made the choice to place them in `matrix` vs `decomp` :)

Reply:
No real thought was put into here. They just need to exist. If you have a preference, please feel free to recommend something. Happy to make the changes. Maybe I'll just move them all to `decomp` to ensure maximum differentiability.

Reply:
I moved all of them to `decomp` except for `MultiControlledX`. I think this is the correct temporary move as this preserves current behaviour and allows us time to work on a context sensitive decomposition strategy. In other words, if we want to differentiate, then we want decomposition. Otherwise, we can use `qml.QubitUnitary`.

Reply:
Sounds good. It would also be worth asking the lightning maintainers if they have a preference for these gates.

Reply:
I think this discussion is definitely worth having, but at a different time. Let's get the data from toml into Catalyst with the current behaviour and then we can update Catalyst to follow more closely what lightning does.